realm/realm-library/src/main/java/io/realm/OrderedRealmCollection.java

```diff
@@ -137,4 +137,14 @@
      * @throws UnsupportedOperationException if the collection is unmanaged.
      */
     boolean deleteLastFromRealm();
+
+    /**
+     * Creates a snapshot from this {@link OrderedRealmCollection}.
+     *
+     * @return the snapshot of this collection.
+     * @see OrderedRealmCollectionSnapshot
+     * @throws java.lang.IllegalStateException if the Realm is closed or the method is called from the wrong thread.
+     * @throws UnsupportedOperationException if the collection is unmanaged.
+     */
+    OrderedRealmCollectionSnapshot<E> createSnapshot();
```

Comment:
YEAH, it is not FREE! But actually the snapshot `TableView` is created through move constructor. So if user doesn't operate on the original Results until the next notification, it is free. Since the bg thread will handover a new `TableView` when next notification arrives.